### PR TITLE
HBASE-23999 [flakey test] TestTableOutputFormatConnectionExhaust

### PR DESCRIPTION
**Pull request by qiaoandxiang**
Do not overwrite the hbase-site.xml which is being used by other tests.

### CODE REVIEW

File: hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionTool.java [hunk 1]
```diff
@@ -60,12 +57,6 @@
   public void setUp() throws Exception {
     this.testUtil.startMiniCluster();
     testUtil.createTable(tableName, HBaseTestingUtility.fam1);
-    String defaultFS = testUtil.getMiniHBaseCluster().getConfiguration().get("fs.defaultFS");
-    Configuration config = HBaseConfiguration.create();
-    config.set("fs.defaultFS", defaultFS);
-    String configPath = this.getClass().getClassLoader()
-      .getResource("hbase-site.xml").getFile();
-    config.writeXml(new FileOutputStream(new File(configPath)));
```
Comment (ndimiduk) [inline hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionTool.java]: @wchevreuil can you comment as to why this test needed to write out it's `hbase-site.xml`? Presumably it was done for a reason and it's not enough to simply not write it out.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 46s |  master passed  |
| +1 :green_heart: |  compile  |   0m 33s |  master passed  |
| -1 :x: |  shadedjars  |   0m 10s |  branch has 7 errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 22s |  hbase-mapreduce in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| -1 :x: |  shadedjars  |   0m  9s |  patch has 7 errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 21s |  hbase-mapreduce in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m 28s |  hbase-mapreduce in the patch passed.  |
|  |   |  24m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1302 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 1765da2b18e7 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9804f73c66 |
| Default Java | 2020-01-14 |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-jdk11-hadoop3-check/output/branch-shadedjars.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-mapreduce.txt |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-jdk11-hadoop3-check/output/patch-shadedjars.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-mapreduce.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/testReport/ |
| Max. process+thread count | 4325 (vs. ulimit of 10000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 45s |  master passed  |
| +1 :green_heart: |  compile  |   0m 27s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  0s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 11s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 18s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 29s |  hbase-mapreduce in the patch passed.  |
|  |   |  33m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1302 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 87cc12047997 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9804f73c66 |
| Default Java | 1.8.0_232 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/testReport/ |
| Max. process+thread count | 4108 (vs. ulimit of 10000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 54s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 48s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 15s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 19s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  16m 15s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  spotbugs  |   1m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  38m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1302 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux b5310c596d39 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9804f73c66 |
| Max. process+thread count | 93 (vs. ulimit of 10000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1302/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**